nc/views.py

```diff
@@ -40,6 +41,37 @@ def search(request):
 class AgencyList(ListView):
     model = Agency
 
+    def get_context_data(self, **kwargs):
+        context = super(AgencyList, self).get_context_data(**kwargs)
+
+        if self.request.method == 'GET' and self.request.GET:
+            form = forms.AgencySearchForm(self.request.GET)
+            if form.is_valid():
+                agency = form.cleaned_data['agency']
+                return redirect('nc:agency-detail', agency.pk)
+        else:
+            form = forms.AgencySearchForm()
+
+        # Once we have the "letters present", we want to be able to iterate
+        # over categorized, sorted lists of agencies. Therefore we create
+        # a dict indexed by first letter.
+        sorted_agencies = {}
+
+        for agency in context['agency_list']:
+            initial = agency.name[:1]
+            if initial not in sorted_agencies:
+                sorted_agencies[initial] = []
+            sorted_agencies[initial].append(agency)
+
+        for key in sorted_agencies:
+            sorted_agencies[key].sort(key=lambda x: x.name)
+            sorted_agencies[key] = get_chunks(sorted_agencies[key])
+
+        sorted_agencies = sorted(sorted_agencies.items())
+
+        return dict(context, **{"sorted_agencies": sorted_agencies,
+                                "agency_form": form, })
```

Comment:
while it may not be one statement, this seems clearer and faster:
```
context['sorted_agencies'] = foo
context['agency_form'] = bar
return context
```